File: 275. H-Index II new.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        long long int l = 0, r = n;
        while (l < r)
        {
            long long int mid = (l+r) >> 1;
            if (citations[mid] >= n-mid)
                r = mid;
            else
                l = mid + 1;
        }
        return n-l;
    }
};

File: 275. H-Index II new.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    var n = citations.length;
    var l = 0;
    var r = n;
    while (l < r)
    {
        var mid = Math.floor((l+r)/2);
        if (citations[mid] >= n-mid)
            r = mid;
        else
            l = mid + 1;
    }
    return n-l;
};
